In [1]:
import numpy as np
from matplotlib import pyplot as plt

#configure plotting
%matplotlib inline
%config InlineBackend.figure_format = 'svg'
import matplotlib
matplotlib.rcParams['figure.figsize'] =  (8,5)
matplotlib.rcParams['text.usetex'] = True
matplotlib.rcParams['font.size'] = 16
matplotlib.rcParams['font.family'] = 'serif'

np.random.seed(0)


In [2]:
X = np.load('twoD_clustering_example.numpy')
print X.shape

(506, 2)


We have a two-dimensional dataset, with 506 data.

Building the model is as simple as importing the GPclust library and calling the class constructor with our data. The argument K specifies how many clusters to use.

In [10]:
import GPclust
m = GPclust.MOG(X, num_clusters=10)
print m.phi

[[ 0.1899072   0.0053279   0.03517011 ...,  0.07245283  0.01314764
   0.40098088]
 [ 0.20134963  0.10387766  0.04287965 ...,  0.26966396  0.07398069
   0.05923953]
 [ 0.08490301  0.15517763  0.04202434 ...,  0.06451708  0.02897642
   0.28431159]
 ..., 
 [ 0.15220887  0.17539786  0.08623784 ...,  0.15292876  0.06494828
   0.03002559]
 [ 0.11378556  0.04689644  0.07583209 ...,  0.02729766  0.02676035
   0.01520369]
 [ 0.03033171  0.12708498  0.00995436 ...,  0.08080533  0.37354209
   0.11119468]]


In [6]:
from pprint import pprint

In [9]:
pprint (vars(m))

{'D': 2,
 'H': 976.8788320755078,
 'Hgrad': array([[ 1.70464333,  3.06853847,  2.48995769, ...,  3.62005289,
         3.57191453,  3.05809718],
       [ 2.78576402,  1.47553409,  2.16876987, ...,  3.13496586,
         2.61673989,  3.78390333],
       [ 5.79613854,  2.58953013,  2.37871253, ...,  3.43033257,
         1.71036951,  1.77378995],
       ..., 
       [ 4.36053297,  0.91658231,  2.36653061, ...,  3.13213606,
         1.4152088 ,  3.41756989],
       [ 3.73982472,  2.16553176,  2.28407991, ...,  3.49759044,
         3.61722014,  2.03856958],
       [ 2.01160654,  4.01134643,  1.76217876, ...,  4.52506454,
         2.1601924 ,  1.88575401]]),
 'N': 506,
 'S0': array([[ 0.001,  0.   ],
       [ 0.   ,  0.001]]),
 'S0_halflogdet': -3.4538776394910689,
 'Sns': array([[[ 0.4260147 ,  0.43768255,  0.49257813,  0.42062048,  0.47318186,
          0.44527465,  0.40184514,  0.48261888,  0.4985085 ,  0.45320676],
        [ 0.09786835,  0.09681348,  0.12464989,  0.09390415,  0.11831129,
 

In [11]:
import GPflow

In [12]:
# build a very simple data set:
N = 12
X = np.random.rand(N,1)
Y = np.sin(12*X) + 0.66*np.cos(25*X) + np.random.randn(N,1)*0.1 + 3

In [13]:
#build the GPR object
k = GPflow.kernels.Matern52(1, lengthscales=0.3)
meanf = GPflow.mean_functions.Linear(1,0)
m = GPflow.gpr.GPR(X, Y, k, meanf)
m.likelihood.variance = 0.01

In [15]:
pprint (vars(m))

{'X': <GPflow.param.DataHolder object at 0x1165aecd0>,
 'Y': <GPflow.param.DataHolder object at 0x116592490>,
 '_free_vars': <tf.Tensor 'Placeholder_2:0' shape=<unknown> dtype=float64>,
 '_name': 'model',
 '_needs_recompile': True,
 '_parent': None,
 '_session': <tensorflow.python.client.session.Session object at 0x1165aef90>,
 '_tf_mode': False,
 'kern': <GPflow.kernels.Matern52 object at 0x1165ae950>,
 'likelihood': <GPflow.likelihoods.Gaussian object at 0x1165ae350>,
 'mean_function': <GPflow.mean_functions.Linear object at 0x1165ae8d0>,
 'num_latent': 1}
